app/models/movie_actor.rb

```diff
@@ -0,0 +1,5 @@
+class MovieActor < ApplicationRecord
+  belongs_to :movie
+  belongs_to :actor
+  has_many :studios, through: :movie
```

Comment:
This is what I mean by over-joining - we already have an association to studios for an actor through movie, a `MovieActor` instance is just the relationship between one movie and one actor. That relationship shouldn't really ever need to be associated to one other model, and if it does, it can reach it via one of the `belongs_to` associations already.